AdminServer/appscale/admin/__init__.py

```diff
@@ -514,12 +718,21 @@ def main():
     hosts=','.join(appscale_info.get_zk_node_ips()),
     connection_retry=ZK_PERSISTENT_RECONNECTS)
   zk_client.start()
+  version_update_lock = zk_client.Lock(constants.VERSION_UPDATE_LOCK_NODE)
```

Comment:
The ZooKeeper lock recipe creates ephemeral nodes. If the AdminServer crashes, the lock will be released automatically.